pkg/cableengine/syncer/syncer.go

```diff
@@ -171,3 +175,15 @@ func (i *GatewaySyncer) generateGatewayObject() (*v1.Gateway, error) {
 	klog.V(log.TRACE).Infof("generateGatewayObject: %+v", gateway)
 	return &gateway, nil
 }
+
+// CleanupOwnEntry removes this Gateway entry from the k8s API, it does not
+// propagate error up because it's a termination function that we also provide externally
+func (s *GatewaySyncer) CleanupOwnEntry() {
+	hostName := s.engine.GetLocalEndpoint().Spec.Hostname
+	err := s.client.Delete(hostName, &metav1.DeleteOptions{})
+	if errors.IsNotFound(err) {
```

Comment:
We should check and report any error.

Reply:
Ah, correct, we are missing the other errors.

Reply:
Actually what I tried to do initially was : `!errors.IsNotFound(err) ` 

not logging the "not found" which means it was already deleted...